internal/cephfs/controllerserver.go

```diff
@@ -83,19 +163,51 @@ func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 	}
 	// TODO need to add check for 0 volume size
 
-	vID, err := checkVolExists(ctx, volOptions, secret)
+	parentVol, pvID, sID, err := checkContentSource(ctx, req, cr)
 	if err != nil {
-		return nil, status.Error(codes.Internal, err.Error())
+		return nil, err
 	}
+	var ecip ErrCloneInProgress
 
+	vID, err := checkVolExists(ctx, volOptions, parentVol, pvID, sID, cr)
+	if err != nil {
+		if errors.As(err, &ecip) {
+			return nil, status.Error(codes.Aborted, err.Error())
+		}
+		return nil, status.Error(codes.Internal, err.Error())
+	}
 	// TODO return error message if requested vol size greater than found volume return error
 
 	if vID != nil {
+		if sID != nil || pvID != nil {
+			// while cloning the volume the size is not populated properly to the new volume now.
+			// it will be fixed in cephfs soon with the parentvolume size. Till then by below
+			// resize we are making sure we return or satisfy the requested size by setting the size
+			// explictly
+			err = resizeVolume(ctx, volOptions, cr, volumeID(vID.FsSubvolName), volOptions.Size)
+			if err != nil {
+				if err = purgeVolume(ctx, volumeID(vID.FsSubvolName), cr, volOptions, false); err != nil {
+					klog.Errorf(util.Log(ctx, "failed to delete volume %s: %v"), requestName, err)
+					// All errors other than ErrVolumeNotFound should return an error back to the caller
+					if !errors.Is(err, ErrVolumeNotFound) {
+						return nil, status.Error(codes.Internal, err.Error())
+					}
+				}
+				errUndo := undoVolReservation(ctx, volOptions, *vID, secret)
+				if errUndo != nil {
+					klog.Warningf(util.Log(ctx, "failed undoing reservation of volume: %s (%s)"),
+						requestName, errUndo)
+				}
+				klog.Errorf(util.Log(ctx, "failed to expand volume %s: %v"), volumeID(vID.FsSubvolName), err)
+				return nil, status.Error(codes.Internal, err.Error())
```

Comment:
This will panic as the err is used for purgeVolume if the purgeVolume err is nil

Reply:
Done.. thanks..